ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -431,6 +438,12 @@ private void updateChunkingAndSizeMetricsOnSuccessfulGet() {
     private BlobId chunkBlobId;
     // whether the operation on the current chunk has completed.
     private boolean chunkCompleted;
+    // refers to decrypt callback Result after decryption, if applicable
+    protected DecryptJob.DecryptJobResult decryptCallbackResult;
```

Comment:
I don't think it is required. If not, decrypt callback processing will be done in next poll(), as we rely on decryptCallbackResultAvailable to go about. 

Btw, I have some confusion on something similar. 
I am not sure if causality will be honored in diff thread contexts. For eg, consider 2 variables, var1, var2. 

Thread 1 : 
if(var2 == true)
// check for var1 value

Thread 2:
set var1 value
set var2 = true.

In this case, I assume since thread1 got the new value of var2, it should have got var1 as well since var1 was set before var2. I am not 100% sure. Will do some study and get back to you.